.github/workflows/merge-check.yml

```
@@ -29,7 +29,7 @@ jobs:
       - name: "Set up Java"
         uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4
         with:
-          distribution: 'temurin'
+          distribution: ${{ (runner.os == 'macOS' && runner.arch == 'ARM64') && 'zulu' || 'temurin' }}
```

Easier and we don't really care that much about this.
```suggestion
          distribution: 'zulu'
```

---

> Left Ubuntu intact, as it's pre-installed!

Using temurin is a no-op (aliasing env vars), while Zulu is always a network request and unzip.

---

Alternative if you don't like the ternary, we can expand the matrix (quite simple).

---

Then we can merge this :)